rust-runtime/aws-smithy-http-server/src/extension.rs

```diff
@@ -39,6 +39,24 @@ use async_trait::async_trait;
 use axum_core::extract::{FromRequest, RequestParts};
 use std::ops::Deref;
 
+/// Extension type used to store the Smithy model namespace.
+#[derive(Debug, Clone)]
+pub struct ExtensionNamespace(pub &'static str);
+
+/// Extension type used to store the Smithy operation name.
+#[derive(Debug, Clone)]
+pub struct ExtensionOperationName(pub &'static str);
+
+/// Extension type used to store the type of framework error caught during exection.
+/// These are unmodeled error, defined in the runtime crates.
+#[derive(Debug, Clone)]
+pub struct ExtensionUnmodeledError(pub String);
```

Comment:
Maybe `ExtensionRejection` is a better name? We've been calling rejections the "errors" returned by the framework. I don't like calling them errors because dealing with malformed requests and rejecting them is frequent enough to be considered part of normal execution flow.

Reply:
Yeah, good suggestion, I'll fix this. Naming things is hard 😛

Reply:
Refactored, thanks for the suggestion!